### PR TITLE
edit-json-file: add options argument for set

### PR DESCRIPTION
**Pull request by bmish**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

The `set` function supports a new options argument. I imported the types for this object from `set-value` since the options are just passed through to that package.

* https://github.com/IonicaBizau/edit-json-file/pull/51
* https://github.com/IonicaBizau/edit-json-file/releases/tag/1.7.0
* https://github.com/IonicaBizau/edit-json-file#setpath-value-options
* https://github.com/jonschlinkert/set-value#options

CC: @flucivja @lukasMega @IonicaBizau @tamslinn

### CODE REVIEW

File: types/edit-json-file/index.d.ts [hunk 1]
```diff
@@ -22,7 +25,7 @@ declare namespace editJsonFile {
         /** Get value at path. */
         get(path?: string): any;
         /** Set value at path. */
-        set(path: string, value: any): JsonEditor;
+        set(path: string, value: any, options?: SetValueOptions): JsonEditor;
```
Comment (martin-badin) [inline types/edit-json-file/index.d.ts:28]: I think `set.Options` is enough
```suggestion
        set(path: string, value: any, options?: set.Options): JsonEditor;
```

**Review by martin-badin — CHANGES_REQUESTED** [review]

File: types/edit-json-file/index.d.ts (hunk 1 above)
Comment (Twixes) [reply types/edit-json-file/index.d.ts:28]: Yup.

Comment (bmish) [reply types/edit-json-file/index.d.ts]: Good catch. Thought that wasn't exported at first but turns out it is available for me to use. Updated.

**Review by Twixes — APPROVED** [review]

**Review by martin-badin — APPROVED** [review]
Looks good

**Pull request merged**